modules/azure/compute.go

```diff
@@ -8,97 +8,451 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-// GetVirtualMachineClientE is a helper function that will setup an Azure Virtual Machine client on your behalf.
-func GetVirtualMachineClientE(subscriptionID string) (*compute.VirtualMachinesClient, error) {
-	// Validate Azure subscription ID
-	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+// VirtualMachineExists indicates whether the specifcied Azure Virtual Machine exists.
+// This function would fail the test if there is an error.
+func VirtualMachineExists(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) bool {
+	exists, err := VirtualMachineExistsE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// VirtualMachineExistsE indicates whether the specifcied Azure Virtual Machine exists.
+func VirtualMachineExistsE(vmName string, resGroupName string, subscriptionID string) (bool, error) {
+	// Get VM Object
+	_, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// GetVirtualMachineNics gets a list of Network Interface names for a specifcied Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineNics(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) []string {
+	nicList, err := GetVirtualMachineNicsE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return nicList
+}
+
+// GetVirtualMachineNicsE gets a list of Network Interface names for a specified Azure Virtual Machine.
+func GetVirtualMachineNicsE(vmName string, resGroupName string, subscriptionID string) ([]string, error) {
+
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
 	if err != nil {
 		return nil, err
 	}
 
-	// Create a VM client
-	vmClient := compute.NewVirtualMachinesClient(subscriptionID)
+	vmNICs := *vm.NetworkProfile.NetworkInterfaces
+	if len(vmNICs) == 0 {
+		// No NIC present
+		return nil, nil
+	}
 
-	// Create an authorizer
-	authorizer, err := NewAuthorizer()
+	// Get the Names of the attached NICs
+	nics := make([]string, len(vmNICs))
+
+	for i, nic := range vmNICs {
+		nics[i] = GetNameFromResourceID(*nic.ID)
+	}
+	return nics, nil
+}
+
+// GetVirtualMachineNicCount gets the Network Interface count of the specified Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineNicCount(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) int {
```

Comment:
NIT: Is this function necessary? Or can one do `len(GetVirtualMachineNics(t, vmName, resGroupName, subscriptionID))` to get the count?

Reply:
Reused, good point!